apollo-compiler/src/main/kotlin/com/apollostack/compiler/ir/TypeDeclaration.kt

```diff
@@ -0,0 +1,22 @@
+package com.apollostack.compiler.ir
+
+import com.squareup.javapoet.TypeSpec
+import javax.lang.model.element.Modifier
+
+data class TypeDeclaration(
+    val kind: String,
+    val name: String,
+    val values: List<TypeDeclarationValue>?) {
+
+  fun toTypeSpec(): TypeSpec {
+    if (kind == "EnumType") {
+      val builder = TypeSpec.enumBuilder(name).addModifiers(Modifier.PUBLIC)
+      values?.forEach {
```

Comment:
if you have null `values`, you'll end up getting an empty enum, which probably won't compile